### ISSUE TITLE
Open Database dialog

### ISSUE THREAD

**Issue by jmucchiello** [sqlitebrowser/sqlitebrowser#659]
Would it be possible to add file types to the preferences so that the choices of file types is not just *.db and *.*? A low priority item, I realize. But if somewhere in the preference there were a place to add additional file types that would show up in the file open/save dialogs, that would be great. 

Maybe even make those items open automatically in the shell. (lower priority)

**Comment by justinclift** [comment]
Sure.  In theory :smile:, they're pretty easy to add.  The list that's supposed to be showing up at the moment is this:

&nbsp; &nbsp; https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/FileDialog.h#L20

Though I kind of remember there have been issues on some OS's at times.  eg they don't show up correctly on some OSX versions, so we disabled them all there.  Made it *.* for Mac.

Which extra extensions do you reckon would be useful? :smile:

**Comment by jmucchiello** [comment]
I would prefer an editable list of extensions somewhere in the preference dialog, which is nowhere near as simple as editing a .h file, I know. But. I use various extensions for various different projects and I doubt they are common with other users. *.PACKAGE, *.CFG, *.OBJECT. See, I doubt many other people would need these.

**Comment by justinclift** [comment]
Yep, sounds like the kind of thing that belongs in the Preferences dialog.

Shouldn't be super hard to code up. :smile: @revolter any interest?

Actually, just in case... @jmucchiello do you have C++/Qt programming skill (or interest to learn)? 😁

**Comment by jmucchiello** [comment]
Only 25+years of C/C++ experience (no Qt experience but I understand the gist of it) but I lack the time to work on the browser. If no one gets around to it in a few months to a year, I might find some time. I agree, though, it is probably a good "get your feet wet" project for someone new to the your code base.

**Comment by justinclift** [comment]
Yeah, time is always the thing that's hard to come by. :smile:

**Comment by justinclift** [comment]
@jmucchiello Does this seem like it would cover all the needed bits?

![edit file extensions concept 1](https://cloud.githubusercontent.com/assets/406299/16976118/5978739a-4e44-11e6-92d4-f2fbb3b6a4c0.png)

It's just a [quick concept mock up in Balsamiq](https://balsamiq.com/free), so it's easy to change elements in if something needs changing. :smile:

**Comment by jmucchiello** [comment]
It doesn't need to be prefilled with my needs. Just prefill it with the current defaults.

Databases (*.db)

And then the dialog also adds All Files (*.*) and if someone wants to delete *.db they can. If they want to add new ones they can. And the up and down arrows are also cool.

**Comment by revolter** [comment]
It's a little more work that the default row type preference feature, and as you know, I don't have time now :( So if no one gets it done in 3 weeks, I'll start working on it.

**Comment by justinclift** [comment]
@jmucchiello Like this?

![edit file extensions concept 2](https://cloud.githubusercontent.com/assets/406299/16976465/f4e025ba-4e46-11e6-88ec-0def11e1ed40.png)

@revolter Yeah, it's probably yours or @innermous's.  After I get #480 done, I have an idea of other stuff I need to focus on already. :smile:

**Comment by chrisjlocke** [comment]
I would change the order of the OK and Cancel buttons... but only because I'm used to Windows....

**Comment by revolter** [comment]
@chrisjlocke, Qt automatically puts the buttons in the OS's default order, so they won't show up exactly like in the mockup on Windows.

**Comment by chrisjlocke** [comment]
Good ol' Qt! Saving the day again, one dialog window at a time!
Thanks for the clarification.  One needs that on a Thursday.

**Comment by GortiZ6** [comment]
I took the care to implement it, let me know if this is fine or you prefer something different :)
[0001-Implemented-feature-request-659.zip](https://github.com/sqlitebrowser/sqlitebrowser/files/1563396/0001-Implemented-feature-request-659.zip)

**Comment by mgrojo** [comment]
Great! I've applied it and for me it's what is expected, I'd only propose some minor adjustments.  I think it would be ideal that you opened a pull request so it is analysed and discussed. But I'm the last one to have arrived this team, maybe it's not needed :wink: Would you need help for that? I may open it in your name.

I've managed to broken it, that would be first needed change :smile:
` ASSERT failure in QList<T>::operator[]: "index out of range", file /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h, line 518
Abortado (`core' generado)
`

This was removing the last added row.

**Comment by justinclift** [comment]
@GortiZ6 Thank you. :smile:

For us, the ideal approach is to create a "Pull Request" with the code you're proposing.  There are a few tutorials on GitHub for how to do this, and other people around the net have written tutorials too.

That being said, that's just a preference.  If you're wanting to contribute code and aren't yet all that familiar with GitHub we're pretty relaxed about stuff and can help guide you. :smile:

**Comment by GortiZ6** [comment]
Thanks for the feedback. I'll fix the bug and submit a pull request on Monday :)

**Comment by justinclift** [comment]
Awesome. :smile:

**Comment by justinclift** [comment]
@jmucchiello This request has just been implemented by @GortiZ6, and will be in tomorrow's nightly builds (onwards).  Any interest in testing it out, and making sure it works well for you? :smile:

If so, the nightly builds are here:

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/latest/
